pkg/network/node/networkpolicy.go

```diff
@@ -975,15 +1005,11 @@ func (np *networkPolicyPlugin) handleAddOrUpdatePod(obj, old interface{}, eventT
 	}
 
 	np.lock.Lock()
-	defer func() {
-		otx := np.node.oc.NewTransaction()
-		otx.DeleteFlows("table=27, cookie=1/-1, ip, nw_src=%s", pod.Status.PodIP)
-		otx.DeleteFlows("table=80, cookie=1/-1, ip, nw_src=%s", pod.Status.PodIP)
-		otx.Commit()
-
-		np.lock.Unlock()
-	}()
-
+	defer np.lock.Unlock()
+	if np.localPodIPs[pod.UID] != "" && pod.Status.PodIP != "" {
+		// cleanup local pod ip once pod.Status.PodIP is set
+		delete(np.localPodIPs, pod.UID)
```

Comment:
we should `delete(np.localPodIPs, pod.UID)` in `handleDeletePod` too, or else we might leak items if pod creation fails in certain ways